lib/screens/NewPatientScreen.dart

```diff
@@ -221,10 +230,23 @@ class _NewPatientFormState extends State<_NewPatientForm> {
   Widget _stickerNumberQuestion() {
     return _makeQuestion('Sticker Number',
       child: TextFormField(
+        decoration: InputDecoration(
+          prefixText: 'PEBRA_',
+        ),
+        keyboardType: TextInputType.number,
+        inputFormatters: [
+          WhitelistingTextInputFormatter.digitsOnly,
+          LengthLimitingTextInputFormatter(3),
+        ],
         controller: _stickerNumberCtr,
+        onEditingComplete: () {
+          _stickerNumberCtr.text = _formatStickerNumber(_stickerNumberCtr.text);
+        },
```

Comment:
I think this does not do anything. Setting the `_stickerNumberCtr.text` value tries to change the text in the UI. However, this won't work because `_formatStickerNumber` prepends "PEBRA_" and "PEBRA_" cannot be inserted into this TextFormField because it's too long (the TextFormField is limited by 3 characters) and also because it contains non-digit characters. This is also why, when saving the patient, the sticker number will just consist of the number without the "PEBRA_" prefix.

Calling `_formatStickerNumber(_stickerNumberCtr.text)` in the `_onSubmitForm` method (line 683) should fix this.